vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -345,15 +345,20 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
           let firstPage = Math.max(0, firstNeededPage);
           let lastPage =
             parentKey !== root ? lastNeededPage : Math.min(lastNeededPage, Math.floor(grid.size / grid.pageSize));
+          let pageCount = lastPage - firstPage + 1;
+          let requestedItemCount = pageCount * grid.pageSize;
+          if ((!parentKey || parentKey === root) && (pageCount > 10 || requestedItemCount > 500)) {
+            throw 'Attempted to fetch more items from server than allowed in one go. ' +
+            'Maximum allowed page count is 10, maximum allowed item count is 500.';
```

Comment:
The problem is caused by using `setAllRowsVisible(true)` together with too many items.

The error message should be more informative about what actions the user should take to fix the issue:
1. Consider not using `setAllRowsVisible(true)` when you have a large amount of items (not only to cover this issue but also to avoid performance bottlenecks resulting from rendering thousands of DOM items).
2. If for some reason this is not an option, increase the page size of the grid so that rendering every item at once doesn't result in a request for over 10 pages